rag_app/ingest.py

```diff
@@ -31,21 +41,18 @@ def batch_chunks(chunks, batch_size=20):
 def chunk_text(
     documents: Iterable[Document], window_size: int = 1024, overlap: int = 0
 ):
-    id = 0
     for doc in documents:
         for chunk_num, start_pos in enumerate(
-            range(0, len(doc.text), window_size - overlap)
+            range(0, len(doc.content), window_size - overlap)
         ):
-            # TODO: Fix up this and use a Lance Model instead - have reached out to the team to ask for some help
             yield {
-                "id": id,
                 "doc_id": doc.id,
-                "chunk_num": chunk_num,
-                "start_pos": start_pos,
-                "end_pos": start_pos + window_size,
-                "text": doc.text[start_pos : start_pos + window_size],
+                "chunk_id": chunk_num+1,
+                "text": doc.content[start_pos : start_pos + window_size],
+                "post_title": doc.metadata["title"],
+                "publish_date": datetime.strptime(doc.metadata["date"], "%Y-%m"),
```

Comment:
We validate that the metadata has a valid date object already

```
class Document(BaseModel):
    id: str
    content: str
    filename: str
    metadata: dict[str, Union[str, List[str]]]

    @field_validator('metadata')
    @classmethod
    def metadata_must_contain_a_valid_datestring(cls, v: dict[str, Union[str, List[str]]]):
        try:
            datetime.strptime(v["date"], "%Y-%m")
        except Exception as e:
            raise ValueError(
                f"Date format must be YYYY-MM (Eg. 2024-10). Unable to parse provided date of {v['date']} "
            )

        return v
```

Perhaps we should be setting a date property on the `Document` object instead of storing it as a metadata dict?